skbase/testing/utils/deep_equals.py

```diff
@@ -105,6 +119,23 @@ def _is_pandas(x):
         return False
 
 
+def _is_npndarray(x):
```

Comment:
Logic looks a bit weird here. Shouldn't we be checking that the name of the class includes "ndarray". 

Also agree with the CodeQL check. It seems like good practice to inlude the else False condition (even though None is returned by default and it won't be interpretted as True).

Reply:
yes, you are right - not sure what I was thinking. Replaced the logic with the "obvious" check